tests/mbed/enc/enc.c

```diff
@@ -98,85 +97,73 @@ static oe_result_t _syscall_hook(
             const int flags = (const int)arg2;
             if (flags == O_RDONLY)
             {
-                syscall_args_t* args;
-                args = (syscall_args_t*)oe_host_malloc(sizeof(syscall_args_t));
-                args->path = oe_host_strdup((const char*)arg1);
-                args->flags = (int)arg2;
-                args->mode = (int)arg3;
-                oe_call_host("mbed_test_open", args);
-                *ret = args->fd;
-                oe_host_free(args->path);
-                oe_host_free(args);
-                result = OE_OK;
+                int rval = 0;
+                result =
+                    mbed_test_open(&rval, (char*)arg1, (int)arg2, (mode_t)arg3);
+                *ret = rval;
             }
             break;
         }
         case SYS_read:
         {
-            syscall_args_t* args;
-            args = (syscall_args_t*)oe_host_malloc(sizeof(syscall_args_t));
-            char* enc_buf = (char*)arg2;
-            char* host_buf = (void*)oe_host_malloc(arg3);
-            args->ptr = (void*)host_buf;
-            args->fd = (int)arg1;
-            args->len = (int)arg3;
-            oe_call_host("mbed_test_read", args);
-
-            if ((args->ret) > 0)
-                oe_memcpy(enc_buf, host_buf, arg3);
-            *ret = args->ret;
+            ssize_t rval = 0;
+            const size_t buf_len = (size_t)arg3;
+            char* host_buf = (char*)oe_host_malloc(buf_len);
+            result = mbed_test_read(&rval, (int)arg1, host_buf, buf_len);
+            if (rval > 0)
+            {
+                char* enc_buf = (char*)arg2;
+                oe_memcpy(enc_buf, host_buf, buf_len);
+            }
+            *ret = (int)rval;
             oe_host_free(host_buf);
-
-            oe_host_free(args);
-            result = OE_OK;
             break;
         }
         case SYS_readv:
         {
-            syscall_args_t* args;
-            args = (syscall_args_t*)oe_host_malloc(sizeof(syscall_args_t));
-            struct iovec* iov = (struct iovec*)arg2;
             int i;
-            struct iovec* iov_host =
-                (struct iovec*)oe_host_malloc(sizeof(struct iovec) * (int)arg3);
-            for (i = 0; i < (int)arg3; i++)
+            struct iovec* iov = (struct iovec*)arg2;
+            const int iovcnt = (int)arg3;
+            struct iovec* host_iov = (struct iovec*)oe_host_malloc(
+                sizeof(struct iovec) * (unsigned long)iovcnt);
+
+            for (i = 0; i < iovcnt; ++i)
             {
-                iov_host[i].iov_base = (void*)oe_host_malloc(iov[i].iov_len);
-                iov_host[i].iov_len = (size_t)iov[i].iov_len;
+                host_iov[i].iov_base = (void*)oe_host_malloc(iov[i].iov_len);
```

Comment:
@EMumau Can you think of how the EDL language can be extended to support some of these scenarios better?
MSR mentioned that users might look at our test code and pick up coding practices. So it would be nice to extend the language to support constructs better.

Reply:
I will have to think on that one.  It is easy to think of cleaner ways to use the enclave.